kernel/src/schema/mod.rs

```diff
@@ -458,6 +458,73 @@ impl StructType {
     }
 }
 
+impl IntoIterator for StructType {
+    type Item = StructField;
+    type IntoIter = StructFieldIntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StructFieldIntoIter {
+            inner: self.fields.into_values(),
+        }
+    }
+}
+
+impl<'a> IntoIterator for &'a StructType {
+    type Item = &'a StructField;
+    type IntoIter = StructFieldRefIter<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StructFieldRefIter {
+            inner: self.fields.values(),
+        }
+    }
+}
+
+pub struct StructFieldIntoIter {
+    inner: indexmap::map::IntoValues<String, StructField>,
+}
+
+impl Iterator for StructFieldIntoIter {
+    type Item = StructField;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.inner.size_hint()
+    }
+}
+
+impl ExactSizeIterator for StructFieldIntoIter {
```

Comment:
I think similarly we could implement [`DoubleEndedIterator`](https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html) since `Values` does too? though can def just take that as a follow-up. `FusedIterator` is an easy one and is also similar.

Reply:
Added! I'm seeing the value of the DoubleEndedIterator, but haven't heard of the FusedIterator before. Not sure where I would use that one, but I've added it anyway. Thanks for the suggestion 👍

Reply:
yea IIRC it basically just means if the iterator returns `None` then it will only ever return `None` after that